packages/store/addon/-private/system/references/belongs-to.js

```diff
@@ -22,13 +23,18 @@ import Reference, { INTERNAL_MODELS } from './reference';
  @extends Reference
  */
 export default class BelongsToReference extends Reference {
-  constructor(store, parentInternalModel, belongsToRelationship, key) {
-    super(store, parentInternalModel);
+  constructor(store, parentIMOrIdentifier, belongsToRelationship, key) {
+    super(store, parentIMOrIdentifier);
     this.key = key;
     this.belongsToRelationship = belongsToRelationship;
     this.type = belongsToRelationship.relationshipMeta.type;
-    this.parent = parentInternalModel.recordReference;
-    this.parentInternalModel = parentInternalModel;
+    if (RECORD_ARRAY_MANAGER_IDENTIFIERS) {
+      this.parent = internalModelFactoryFor(store).peek(parentIMOrIdentifier).recordReference;
+      this.parentInternalModel = internalModelFactoryFor(store).peek(parentIMOrIdentifier);
+    } else {
+      this.parent = parentIMOrIdentifier.recordReference;
+      this.parentInternalModel = parentIMOrIdentifier;
+    }
```

Comment:
If the goal is to remove references to the internal model property as well, I can make this `this.parentIdentifier` and lookup by identifier in each corresponding method.

Reply:
If we have a path of fully getting away from internal model, we probably want to bundle this up together, landing just the lookup part might be bad for perf in hot paths